src/main/java/seedu/address/logic/commands/schedule/AddScheduleCommand.java

```diff
@@ -76,8 +75,9 @@ public CommandResult execute(Model model) throws CommandException {
         if (model.hasConflictingInterview(toAdd)) {
             throw new CommandException(MESSAGE_CONFLICTING_INTERVIEW);
         }
-        model.setCandidate(candidateToInterview, editedCandidate);
         model.addInterview(toAdd);
+        Candidate editedCandidate = candidateToInterview.triggerInterviewStatusScheduled();
+        model.setCandidate(candidateToInterview, editedCandidate);
```

Comment:
@lzan98 hmm leon this fixes?

Reply:
Yeap just shifting! Because while doing the sequence diagram I realised it doesn't make much sense to trigger the interview status BEFORE adding the interview to the model from a logical point of view

Reply:
Noted! I get what you mean. Thanks for clarifying leon!! 😄